src/platform/forms-system/src/js/fields/TitleField.jsx

```diff
@@ -1,18 +1,49 @@
 import React from 'react';
+
+import PropTypes from 'prop-types';
 import classNames from 'classnames';
 
-export default function TitleField({ id, title }) {
+export default function TitleField({ id, title, formContext }) {
   const isRoot = id === 'root__title';
-
-  const classes = classNames('schemaform-block-title', {
+  const classes = classNames('schemaform-block-title testing', {
     'schemaform-block-subtitle': !isRoot,
   });
+  const { onReviewPage } = formContext;
+  let reviewTitle;
+
+  const isStringTitle = typeof title === 'string';
+  const isEmptyStringTitle = isStringTitle && title.trim() === '';
 
-  const isEmptyTitle = typeof title === 'string' && title.trim() === '';
+  // Handle title scenarios:
+  // If title is a React element with an <h3>,
+  // it should be made <h4> for the Review page.
+  if (isStringTitle) {
```

Comment:
- if we go with this logic, could you make this so there is only a conditional for our explicit scenario?  for example if (onReviewPage && title.type ==='h3') ... then do what we want, otherwise use default. 
- and then for the existing isEmptyTitle scenario, do we still want to return null instead of a legend with null?